torcheeg/datasets/module/emotion_recognition/faced.py

```diff
@@ -0,0 +1,340 @@
+                 num_baseline: int = 3,
+                 online_transform: Union[None, Callable] = None,
+                 offline_transform: Union[None, Callable] = None,
+                 label_transform: Union[None, Callable] = None,
+                 before_trial: Union[None, Callable] = None,
+                 after_trial: Union[Callable, None] = None,
+                 after_subject: Union[Callable, None] = None,
+                 io_path: Union[None, str] = None,
+                 io_size: int = 1048576,
+                 io_mode: str = 'lmdb',
+                 num_worker: int = 0,
+                 verbose: bool = True):
+        if io_path is None:
+            io_path = get_random_dir_path(dir_prefix='datasets')
+
+        # pass all arguments to super class
+        params = {
+            'root_path': root_path,
+            'chunk_size': chunk_size,
+            'overlap': overlap,
+            'num_channel': num_channel,
+            'num_baseline': num_baseline,
+            'online_transform': online_transform,
+            'offline_transform': offline_transform,
+            'label_transform': label_transform,
+            'before_trial': before_trial,
+            'after_trial': after_trial,
+            'after_subject': after_subject,
+            'io_path': io_path,
+            'io_size': io_size,
+            'io_mode': io_mode,
+            'num_worker': num_worker,
+            'verbose': verbose
+        }
+        super().__init__(**params)
+        # save all arguments to __dict__
+        self.__dict__.update(params)
+
+    @staticmethod
+    def process_record(file: Any = None,
+                       root_path: str = './Processed_data',
+                       chunk_size: int = 250,
+                       overlap: int = 0,
+                       num_channel: int = 30,
+                       num_baseline: int = 3,
+                       baseline_chunk_size: int = 250,
+                       before_trial: Union[None, Callable] = None,
+                       offline_transform: Union[None, Callable] = None,
+                       **kwargs):
+        file_name = os.path.basename(file)  # an element from file name list, such as 'sub087.pkl'
+        subject_id = int(file_name.split('.')[0][3:]) # get subject_id from 'sub087.pkl', such as 87
+        # derive the given arguments (kwargs)
+        with open(os.path.join(root_path, file_name), 'rb') as f:
+            samples = pkl.load(f, encoding='iso-8859-1') # 28(trials), 32(channels), 30s*250hz(time points)
+
+        write_pointer = 0
+
+        for trial_id in range(len(samples)):
+
+            # extract baseline signals
+            trial_samples = samples[
+                trial_id, :num_channel]  # 30(remove A1 and A2 from 32 channels), 30s*250hz(time points)
+            if before_trial:
+                trial_samples = before_trial(trial_samples)
+
+            trial_baseline_sample = trial_samples[:, :baseline_chunk_size *
```

Comment:
Indeed, some datasets provide baseline signals, such as DEAP, where signals were segmented into 60-second trials with a 3-second pre-trial baseline.

However, please refer to the paper https://www.nature.com/articles/s41597-023-02650-w

> the last 30 seconds of each video clip were selected to capture the maximal emotional responses.

In FACED, all 30-second EEG signals are all trial signals.

Could you please refer to the implementation of the SEED dataset and remove the parts related to baseline signals? https://github.com/torcheeg/torcheeg/blob/main/torcheeg/datasets/module/emotion_recognition/seed.py

Reply:
I understand. [fix: faced dataset has no baseline signals](https://github.com/torcheeg/torcheeg/pull/108/commits/fe0109f9374d1d3e1123956fc0772ba7f6589658) solved it.